cmd/main_test.go

```diff
@@ -267,3 +275,38 @@ var _ = AfterSuite(func() {
 		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Error killing the server during test teardown: %v", err))
 	}
 })
+
+func GetNamespace(k8sClient client.Client, namespace string) (k8sapi.Namespace, error) {
+	ns := &k8sapi.Namespace{}
+	err := k8sClient.Get(context.Background(), client.ObjectKey{Name: namespace}, ns)
+	if err != nil {
+		return k8sapi.Namespace{}, fmt.Errorf("error getting namespace: %w", err)
+	}
+	return ns, nil
+}
+
+var _ = DescribeTable("TestGetNamespacesWithAccess", func(allNamespaces []k8sapi.Namespace, 
+	expectedNs []k8sapi.Namespace) {
+	e := echo.New()
+	cfg, _ := config.GetConfig()
+	clientset, _ := kubernetes.NewForConfig(cfg)
+	req := httptest.NewRequest(http.MethodGet, "/", nil)
+	rec := httptest.NewRecorder()
+	c := e.NewContext(req, rec)
+	c.Request().Header.Set("X-Email", "test@example.com")
+	authCl := clientset.AuthorizationV1()
+
+	actualNs, err := getNamespacesWithAccess(e, c, authCl, allNamespaces)
+
+	Expect(actualNs).To(Equal(expectedNs))
+	Expect(err).NotTo(HaveOccurred(), "Unexpected error testing GetNamespacesWithAccess")
+},
+	Entry(
+		"when access check allows all namespaces",
```

Comment:
`Access check` is an implementation detail. the unit test should not rely on it as the way we'll verify access